src/featurize.py

```diff
@@ -0,0 +1,125 @@
+import numpy as np
+import scipy as sp
+import sklearn
+import sklearn.preprocessing
+import copy
+
+def generate_feature_matrix(data, degree=2):
+
+    """
+    Function to generate a featurized data matrix from a data matrix where each
+    row is a scattering event. We use polynomial feature with arbitrary degree
+    as well as dot product of different ion/electron momenta.
+
+    Parameters
+    ------------
+    data : np.array [todo] How does this work if I pass in a pandas array
+        Array containing a row for each scattering event. We have 15 columns
+        (px, py, pz) for each of 5 scattering products: ion 1, ion 2, neutral,
```

Comment:
Please move the [todo] into a github issue and remove. Also at first glance this function seams to work for an array of arbitrary size. But the doc says it works only for arrays with 15 columns. Which one is correct?

Reply:
I removed the part about pandas dataframes so that we only work with numpy arrays. The function works with an array of arbitrary size, but I was giving the sample of our typical dataset. I clarified this in the doc string.